.github/workflows/golangci-lint.yaml

```
@@ -45,7 +45,27 @@ jobs:
     steps:
       - uses: actions/setup-go@d35c59abb061a4a6fb18e82ac0862c26744d6ab5 # v5.5.0
         with:
-         go-version: "1.23"
+         go-version: "1.24"
          check-latest: true
       - name: govulncheck
         uses: golang/govulncheck-action@b625fbe08f3bccbe446d94fbf87fcc875a4f50ee # v1.0.4
+
+linters:
+  enable:
+    - staticcheck
+
+exclude-rules:
+  - path: pkg/operator/ceph/client/controller_test.go
+    linters:
+      - staticcheck
+    text: "res.Requeue is deprecated"
+
+  - path: pkg/operator/ceph/controller/spec.go
+    linters:
+      - staticcheck
+    text: "v1.Endpoints is deprecated"
+
+  - path: pkg/operator/ceph/object/spec.go
+    linters:
+      - staticcheck
+    text: "v1.Endpoints is deprecated"
```

```suggestion
linters:
  enable:
    - staticcheck

exclude-rules:
  - path: pkg/operator/ceph/client/controller_test.go
    linters:
      - staticcheck
    text: "res.Requeue is deprecated"

  - path: pkg/operator/ceph/controller/spec.go
    linters:
      - staticcheck
    text: "v1.Endpoints is deprecated"

  - path: pkg/operator/ceph/object/spec.go
    linters:
      - staticcheck
    text: "v1.Endpoints is deprecated"
```

Should we do changes here https://github.com/rook/rook/blob/master/.golangci.yaml

---

done. thanks!